sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py

```diff
@@ -391,7 +401,18 @@ async def close(self):
             self.writer.close()
             await self.writer.wait_closed()
             self.writer, self.reader = None, None
-        self.sock = None
+        if self.sock is not None:
+            # Call shutdown first to make sure that pending messages
+            # reach the AMQP broker if the program exits after
+            # calling this method.
+            try:
+                self.sock.shutdown(socket.SHUT_RDWR)
+            except Exception as exc:  # pylint: disable=broad-except
+                # TODO: shutdown could raise OSError, Transport endpoint is not connected if the endpoint is already
+                #  disconnected. can we safely ignore the errors since the close operation is initiated by us.
+                _LOGGER.info("Transport endpoint is already disconnected: %r", exc)
+            self.sock.close()
+            self.sock = None
```

Comment:
copied from sync

Reply:
I don't know if this is necessary? We pass the socket into `open_connection` - doesn't that close it on our behalf when we close the reader and writer?

Reply:
ah, I see. That's prob why this wasn't here before 🙂. Will add a note here.